tests/quantization/torchao/test_torchao.py

```diff
@@ -0,0 +1,358 @@
+                ]
+            )
+
+        for quantization_name, expected_slice in QUANTIZATION_TYPES_TO_TEST:
+            quantization_config = TorchAoConfig(quant_type=quantization_name)
+            self._test_quant_type(quantization_config, expected_slice)
+
+    def test_int4wo_quant_bfloat16_conversion(self):
+        """
+        Tests whether the dtype of model will be modified to bfloat16 for int4 weight-only quantization.
+        """
+        quantization_config = TorchAoConfig("int4_weight_only", group_size=64)
+        quantized_model = FluxTransformer2DModel.from_pretrained(
+            "hf-internal-testing/tiny-flux-pipe",
+            subfolder="transformer",
+            quantization_config=quantization_config,
+            torch_dtype=torch.bfloat16,
+        )
+
+        qlayer = quantized_model.transformer_blocks[0].attn.to_q
+        weight = qlayer.weight
+        self.assertTrue(isinstance(weight, AffineQuantizedTensor))
+        self.assertEqual(weight.quant_min, 0)
+        self.assertEqual(weight.quant_max, 15)
+        self.assertTrue(isinstance(weight.layout_type, TensorCoreTiledLayoutType))
+
+    def test_int4wo_offload(self):
+        """
+        Test if the quantized model int4 weight-only is working properly with cpu/disk offload.
+        """
+
+        device_map_offload = {
+            "time_text_embed": torch_device,
+            "context_embedder": torch_device,
+            "x_embedder": torch_device,
+            "transformer_blocks.0": "cpu",
+            "single_transformer_blocks.0": "disk",
+            "norm_out": torch_device,
+            "proj_out": "cpu",
+        }
+
+        batch_size = 1
+        num_latent_channels = 4
+        num_image_channels = 3
+        height = width = 4
+        sequence_length = 48
+        embedding_dim = 32
+
+        hidden_states = torch.randn((batch_size, height * width, num_latent_channels)).to(
+            torch_device, dtype=torch.bfloat16
+        )
+        encoder_hidden_states = torch.randn((batch_size, sequence_length, embedding_dim)).to(
+            torch_device, dtype=torch.bfloat16
+        )
+        pooled_prompt_embeds = torch.randn((batch_size, embedding_dim)).to(torch_device, dtype=torch.bfloat16)
+        text_ids = torch.randn((sequence_length, num_image_channels)).to(torch_device, dtype=torch.bfloat16)
+        image_ids = torch.randn((height * width, num_image_channels)).to(torch_device, dtype=torch.bfloat16)
+        timestep = torch.tensor([1.0]).to(torch_device, dtype=torch.bfloat16).expand(batch_size)
+
+        with tempfile.TemporaryDirectory() as offload_folder:
+            quantization_config = TorchAoConfig("int4_weight_only", group_size=64)
+            quantized_model = FluxTransformer2DModel.from_pretrained(
+                "hf-internal-testing/tiny-flux-pipe",
+                subfolder="transformer",
+                quantization_config=quantization_config,
+                device_map=device_map_offload,
```

Comment:
We assign a `hf_device_map` attribute to the model too, so we should also check if the `quantized_hf_device_map` matches the expected one.

Reply:
@a-r-r-o-w just checking if this is remaining to be added?

Reply:
Updated the `test_offload` test with a check to verify `hf_device_map` is same as `device_map`. I don't see any `quantized_hf_device_map` when grepping the codebase or searching on github, so not sure what you are referring too. Could you help with this?